worlds/metroidprime/__init__.py

```diff
@@ -0,0 +1,473 @@
+        if self.starting_beam:
+            slot_data["starting_beam"] = self.starting_beam
+
+        return slot_data
+
+        # for the universal tracker, doesn't get called in standard gen
+
+    @staticmethod
+    def interpret_slot_data(slot_data: Dict[str, Any]) -> Dict[str, Any]:
+        # returning slot_data so it regens, giving it back in multiworld.re_gen_passthrough
+        info("Regenerating world for tracker")
+        return slot_data
+
+    def write_spoiler(self, spoiler_handle: TextIO):
+        player_name = self.player_name
+
+        spoiler_handle.write(
+            f"Starting Room({player_name}): {self.starting_room_name}\n"
+        )
+
+        if self.options.randomize_starting_beam:
+            spoiler_handle.write(
+                f"Starting Beam({player_name}): {self.starting_beam}\n"
+            )
+
+        if self.options.elevator_randomization:
+            spoiler_handle.write(f"\n\nElevator Mapping({player_name}):\n")
+            for area, mapping in self.elevator_mapping.items():
+                spoiler_handle.write(f"{area}:\n")
+                for source, target in mapping.items():
+                    spoiler_handle.write(
+                        f"    {ELEVATOR_USEFUL_NAMES[source]} -> {ELEVATOR_USEFUL_NAMES[target]}\n"
+                    )
+
+        if (
+            self.options.door_color_randomization
+            == DoorColorRandomization.option_regional
+        ):
+            assert self.door_color_mapping is not None
+            spoiler_handle.write(f"\n\nDoor Color Mapping({player_name}):\n")
+
+            for area, mapping in self.door_color_mapping.items():
+                spoiler_handle.write(f"{area}:\n")
+                for door, color in mapping.type_mapping.items():
+                    spoiler_handle.write(f"    {door} -> {color}\n")
+
+        elif (
+            self.options.door_color_randomization
+            == DoorColorRandomization.option_global
+        ):
+            assert self.door_color_mapping is not None
+            spoiler_handle.write(f"\n\nDoor Color Mapping({player_name}):\n")
+            for door, color in self.door_color_mapping[
+                MetroidPrimeArea.Tallon_Overworld.value
+            ].type_mapping.items():
+                spoiler_handle.write(f"    {door} -> {color}\n")
+
+        if (
+            self.options.blast_shield_randomization.value
+            != BlastShieldRandomization.option_none
+            or self.options.locked_door_count > 0
+        ):
+            assert self.blast_shield_mapping is not None
+            spoiler_handle.write(f"\n\nBlast Shield Mapping({player_name}):\n")
+            written_mappings: List[List[str]] = []
+
+            for area, mapping in self.blast_shield_mapping.items():
+                spoiler_handle.write(f"{area}:\n")
+                if len(mapping.type_mapping) == 0:
+                    spoiler_handle.write(f"    None\n")

```

Comment:
I would put a `continue` in here to make it extra clear that the code below this point is not expected to be run when this condition branch is hit.

```suggestion
                    spoiler_handle.write(f"    None\n")
                    continue
```